Add container_clean and container_stop steps to Makefile

elsesiy opened the pull request:
Fix #42

elsesiy commented:
Hi @nitishm,
I saw this issue is up for grabs so I quickly fixed it. I think it meets all the requirements stated in the issue but I couldn't figure out what you meant by 
`make container_run should also, first depend on make container followed by make container_stop`. I don't think it's a good idea to always clean (namely remove image and container) after each run so I left it as a task to the user to manually run `container_clean` or `container_stop` after `container_run`.

In general, I feel like there's more room for improvement considering your usage of the full-fledged `golang` base image in `Dockerfile` etc. 

I wrote a [blog post](https://elsesiy.com/blog/containerization-of-golang-applications) a while back in case you're interested in my approach.

Cheers, Jonas

nitishm commented:
@elsesiy thanks for taking up this task. I am all for improvements and learning so please create a PR with the proposed changes and we can get it reviewed and merged. My  `make`-fu isn't the strongest so I would love to learn as well!

elsesiy commented:
@nitishm Do you want me to update this PR or create a new issue in which I suggest the changes and open another PR? Depending on your answers we could merge this now.

nitishm commented:
You can update this PR.

elsesiy commented:
@nitishm Okay I updated the PR accordingly and removed a bunch of things. The final docker image size has been reduced from `1.03GB` to `16MB`. Since travis uses a very broad go versioning `1.x` the build has been running with `1.12.x` for a while and I updated the `Dockerfile` accordingly. I'm sure there's more stuff to take care of but this takes care of the most obvious things.

nitishm commented on Makefile:
```diff
@@ -1,14 +1,14 @@
+CONTAINER_NAME ?= vegeta
 
 SERVER_DIR = cmd/server
-GO  = GO111MODULE=on go
```
So I get that Travis would handle this correctly, but what about running this on a local system. For `go mod` to work we need to set `GO111MODULE` (not sure if this is the case in 1.12 though) right ?

elsesiy replied:
Actually no, as long as `go.mod` is present in the workdir this defaults to `GO111MODULE=on`. More on this can be found [here](https://blog.golang.org/modules2019).
